File: solution/169. Majority Element/169_02.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) { // using Moore Voting Algorithm
        int res = 0, cnt = 0;
        for (int x : nums) {
            if (cnt == 0) {
                res = x;
                cnt++;
            } else {
                res == x ? cnt++ : cnt--;
            }            
        }
        return res;
    }
};

File: solution/169. Majority Element/169_02.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        val, cnt = 0, 0
        for x in nums:
            if cnt == 0:
                val = x
                cnt = 1
            else:
                cnt += 1 if val == x else -1
        return val
    
